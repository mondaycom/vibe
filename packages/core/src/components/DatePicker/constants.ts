export const YEAR_FORMAT = "YYYY";
export const MONTH_FORMAT = "MMMM";
export const DAY_SIZE = 40;
export const WEEK_FIRST_DAY = 1;
