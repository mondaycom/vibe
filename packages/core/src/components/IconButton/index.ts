export { default as IconButton, type IconButtonProps } from "./IconButton";
