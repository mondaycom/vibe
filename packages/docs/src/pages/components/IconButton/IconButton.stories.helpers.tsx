import React from "react";
import { StorybookLink, Tip } from "vibe-storybook-components";

export const TipCheckYourself = () => (
  <Tip title="Check yourself">
    To display icons that do not have actions associated with them, use the{" "}
    <StorybookLink page="Components/Icon" size={StorybookLink.sizes.SMALL}>
      Icon component
    </StorybookLink>
    .
  </Tip>
);

export const TipInfo = () => (
  <Tip>
    If you need to use an icon as a button that opens info dialog with additional information, check out{" "}
    <StorybookLink page="Components/Info" size={StorybookLink.sizes.SMALL}>
      Info
    </StorybookLink>{" "}
    component.
  </Tip>
);

export const TipMenuButton = () => (
  <Tip>
    If you need to use an icon as a button that opens menu next to it, check out{" "}
    <StorybookLink page="Components/MenuButton" size={StorybookLink.sizes.SMALL}>
      Menu button
    </StorybookLink>{" "}
    component.
  </Tip>
);
