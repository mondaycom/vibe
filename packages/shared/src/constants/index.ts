export * from "./keyCodes";
export * from "./sizes";
