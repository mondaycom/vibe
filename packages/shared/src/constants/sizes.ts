export const BASE_SIZES = {
  SMALL: "small",
  MEDIUM: "medium",
  LARGE: "large"
} as const;

export const SIZES = { XXS: "xxs", XS: "xs", ...BASE_SIZES } as const;

export enum BaseSizes {
  SMALL = "small",
  MEDIUM = "medium",
  LARGE = "large"
}

export enum Sizes {
  XXS = "xxs",
  XS = "xs",
  SMALL = "small",
  MEDIUM = "medium",
  LARGE = "large"
}

export type SIZES_VALUES = (typeof SIZES)[keyof typeof SIZES];
