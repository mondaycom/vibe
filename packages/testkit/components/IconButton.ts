import { Locator, Page, test } from "@playwright/test";
import { Button } from "./Button";
import { Menu } from "./Menu";

/**
 * Class representing an icon button that extends the Button class.
 */
export class IconButton extends Button {
  icon: Button;
  menu: Menu | undefined;

  /**
   * Create an IconButton.
   * @param {Page} page - The Playwright page object.
   * @param {Locator} locator - The locator for the IconButton element.
   * @param {string} elementReportName - The name for reporting purposes.
   * @param menuType - The type of menu associated with the button.
   */
  constructor(page: Page, locator: Locator, elementReportName: string, menuType?: Menu) {
    super(page, locator, elementReportName);
    this.icon = new Button(this.page, this.locator, `${this.elementReportName} - Icon`);
    this.menu = menuType;
  }

  /**
   * Select an item from the icon button's menu.
   * @param {string} item - The item to select.
   * @returns {Promise<void>}
   */
  async selectItem(item: string): Promise<void> {
    await test.step(
      `Select ${item} from ${this.elementReportName}`,
      async () => {
        await this.icon.click();
        if (this.menu) {
          await this.menu.selectItem(item);
        }
      },
      { box: false }
    );
  }
}
