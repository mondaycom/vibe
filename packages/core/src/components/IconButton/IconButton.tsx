import React, { AriaAttributes, forwardRef, Fragment, useMemo, useRef } from "react";
import cx from "classnames";
import { noop as NOOP } from "lodash-es";
import useMergeRef from "../../hooks/useMergeRef";
import Tooltip, { TooltipProps } from "../Tooltip/Tooltip";
import Icon from "../Icon/Icon";
import { AddSmall } from "@vibe/icons";
import { getWidthHeight, Size } from "./services/IconButton-helpers";
import { SubIcon, VibeComponent, VibeComponentProps, withStaticProps } from "../../types";
import { getTestId } from "../../tests/test-ids-utils";
import { ComponentDefaultTestId, ComponentVibeId } from "../../tests/constants";
import Button from "../Button/Button";
import { BUTTON_ICON_SIZE, SMALL_BUTTON_ICON_SIZE } from "../Button/ButtonConstants";
import { ButtonColor, ButtonType } from "../Button/Button.types";
import { getStyle } from "../../helpers/typesciptCssModulesHelper";
import styles from "./IconButton.module.scss";

export interface IconButtonProps extends VibeComponentProps {
  /**
   * Callback fired when the button is clicked.
   */
  onClick?: (event: React.MouseEvent) => void;
  /**
   * Class name applied to the button wrapper.
   */
  wrapperClassName?: string;
  /**
   * Class name applied to the icon.
   */
  iconClassName?: string;
  /**
   * The icon displayed inside the button.
   */
  icon?: SubIcon;
  /**
   * The ID of the element that labels this button.
   */
  ariaLabeledBy?: string;
  /**
   * The ARIA label for accessibility.
   */
  ariaLabel?: string;
  /**
   * If true, indicates that the button controls a popup.
   */
  ariaHasPopup?: React.HTMLProps<HTMLButtonElement>["aria-haspopup"];
  /**
   * If true, indicates that the associated popup is open.
   */
  ariaExpanded?: boolean;
  /**
   * The ID of the region controlled by the button.
   */
  ariaControls?: string;
  /**
   * ID of the element describing the button.
   */
  "aria-describedby"?: AriaAttributes["aria-describedby"];
  /**
   * If true, hides the button from assistive technologies.
   */
  "aria-hidden"?: AriaAttributes["aria-hidden"];
  /**
   * Indicates the current "pressed" state of toggle buttons.
   */
  "aria-pressed"?: AriaAttributes["aria-pressed"];
  /**
   * The size of the button.
   */
  size?: Size;
  /**
   * If true, hides the tooltip.
   */
  hideTooltip?: boolean;
  /**
   * Props for the Tooltip component.
   */
  tooltipProps?: Partial<TooltipProps>;
  /**
   * Tooltip content displayed on hover.
   */
  tooltipContent?: string;
  /**
   * The button variant.
   */
  kind?: ButtonType;
  /**
   * If true, the button is in an active state.
   */
  active?: boolean;
  /**
   * The color of the button.
   */
  color?: ButtonColor;
  /**
   * If true, the button is disabled.
   */
  disabled?: boolean;
  /**
   * If disabled, this message will be displayed in the tooltip.
   */
  disabledReason?: string;
  /**
   * If true, the focus indicator is displayed inside the button instead of around it.
   */
  insetFocus?: boolean;
  /**
   * The tab order of the button.
   */
  tabIndex?: number;
  /**
   * If true, a loader replaces the icon.
   */
  loading?: boolean;
}

const IconButton: VibeComponent<IconButtonProps> & {
  sizes?: typeof Button.sizes;
  kinds?: typeof Button.kinds;
  colors?: typeof Button.colors;
} = forwardRef(
  (
    {
      className,
      wrapperClassName,
      iconClassName,
      id,
      icon = AddSmall,
      size = "medium",
      tooltipProps = {} as TooltipProps,
      tooltipContent,
      ariaLabeledBy,
      ariaLabel,
      ariaHasPopup,
      ariaExpanded,
      ariaControls,
      "aria-describedby": ariaDescribedBy,
      "aria-hidden": ariaHidden,
      "aria-pressed": ariaPressed,
      hideTooltip = false,
      kind = "tertiary",
      active,
      disabled = false,
      disabledReason,
      onClick = NOOP,
      color,
      "data-testid": dataTestId,
      insetFocus = false,
      tabIndex,
      loading = false
    }: IconButtonProps,
    ref
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRef(ref, componentRef);

    const overrideTooltipContent = useMemo(
      () => tooltipProps?.content || tooltipContent,
      [tooltipProps?.content, tooltipContent]
    );

    const buttonAriaLabel = useMemo(() => {
      if (ariaLabel) return ariaLabel;
      if (typeof overrideTooltipContent === "string") return overrideTooltipContent;
      return undefined;
    }, [ariaLabel, overrideTooltipContent]);

    const iconSize = useMemo(() => {
      switch (size) {
        case "xxs":
        case "xs":
          return SMALL_BUTTON_ICON_SIZE;
        case "small":
        case "medium":
        case "large":
          return BUTTON_ICON_SIZE;
        default:
          return 24;
      }
    }, [size]);

    const overrideStyle = useMemo(() => {
      let style = {
        justifyContent: "center",
        alignItems: "center",
        padding: 0
      } as React.CSSProperties;

      if (size) {
        style = { ...style, ...getWidthHeight(size) };
      }
      return style;
    }, [size]);

    const calculatedTooltipContent = useMemo(() => {
      if (hideTooltip) return null;
      if (disabled && disabledReason) return disabledReason;
      if (overrideTooltipContent) return overrideTooltipContent as never;
      return ariaLabel;
    }, [hideTooltip, disabled, disabledReason, overrideTooltipContent, ariaLabel]);

    const IconButtonWrapper = wrapperClassName ? "div" : Fragment;
    const iconButtonWrapperProps = useMemo(() => {
      return wrapperClassName ? { className: cx(wrapperClassName, styles.wrapper) } : {};
    }, [wrapperClassName]);

    return (
      <IconButtonWrapper {...iconButtonWrapperProps}>
        <Tooltip
          {...tooltipProps}
          content={calculatedTooltipContent}
          referenceWrapperClassName={styles.referenceWrapper}
        >
          <Button
            onClick={onClick}
            disabled={disabled}
            color={color}
            kind={kind}
            ariaLabeledBy={ariaLabeledBy}
            ariaLabel={buttonAriaLabel}
            ariaHasPopup={ariaHasPopup}
            ariaExpanded={ariaExpanded}
            ariaControls={ariaControls}
            aria-describedby={ariaDescribedBy}
            aria-hidden={ariaHidden}
            aria-pressed={ariaPressed}
            ref={mergedRef}
            id={id}
            data-testid={dataTestId || getTestId(ComponentDefaultTestId.ICON_BUTTON, id)}
            data-vibe={ComponentVibeId.ICON_BUTTON}
            noSidePadding
            active={active}
            className={className}
            style={overrideStyle}
            insetFocus={insetFocus}
            tabIndex={tabIndex}
            loading={loading}
            loaderClassName={cx(styles.loader, getStyle(styles, size))}
          >
            <Icon icon={icon} iconType="svg" iconSize={iconSize} ignoreFocusStyle className={iconClassName} />
          </Button>
        </Tooltip>
      </IconButtonWrapper>
    );
  }
);

export default withStaticProps(IconButton, {
  sizes: Button.sizes,
  kinds: Button.kinds,
  colors: Button.colors
});
