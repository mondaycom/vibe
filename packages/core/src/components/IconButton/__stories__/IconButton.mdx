import { Meta } from "@storybook/blocks";
import IconButton from "../IconButton";
import DialogContentContainer from "../../DialogContentContainer/DialogContentContainer";
import { Add, Bolt, Broom, HighlightColorBucket, Pin, Show } from "@vibe/icons";
import { TipCheckYourself, TipMenuButton } from "./IconButton.stories.helpers";
import * as IconButtonStories from "./IconButton.stories";

<Meta of={IconButtonStories} />

# IconButton

Icon button is a square button contains only icon with no visible text labels used mostly in control bars.

<Canvas of={IconButtonStories.Overview} />

### Import

```js
import { IconButton } from "@vibe/core";
```

## Props

<PropsTable />

## Usage

<UsageGuidelines
  guidelines={[
    <>
      Icon button will always appear with a <StorybookLink page="Components/Tooltip">tooltip</StorybookLink> while
      hovering, defining the icon’s meaning.
    </>,
    "Use an icon button when a user can perform an inline action on this page and there's no room for a default button.",
    "Use icon button when you want to display an active state of a button.",
    "Use a Primary icon button when it's the most important action to take.",
    "Use icon button only when the icons is clear and understandable.",
    "Icon buttons are often used when there are 2 or 3 adjacent icons buttons that perform actions on a single item presented in a row."
  ]}
/>

<TipCheckYourself />

## Accessibility

<UsageGuidelines
  guidelines={[
    <>
      Provide an <code>id</code> for the IconButton to enable proper accessibility associations and unique
      identification.
    </>,
    <>
      Always provide an accessible name using either <code>ariaLabel</code> prop or <code>tooltipContent</code> prop.
      The component automatically uses <code>tooltipContent</code> as the accessible name if no <code>ariaLabel</code>{" "}
      is provided.
    </>,
    <>
      Use <code>ariaLabel</code> prop when you need a different accessible name than the tooltip text, or when the icon
      button doesn't have a tooltip.
    </>,
    <>
      Use <code>ariaLabeledBy</code> prop when an external element provides the accessible name for the icon button.
      Pass the <code>id</code> of that external element.
    </>,
    <>
      Use <code>ariaHasPopup</code> prop when the icon button opens a menu, dialog, or popup.
    </>,
    <>
      Use <code>ariaExpanded</code> prop to indicate when a popup or menu opened by the icon button is currently open (
      <code>true</code>) or closed (<code>false</code>).
    </>,
    <>
      Use <code>ariaControls</code> prop to link the icon button to the element it controls. Pass the <code>id</code> of
      the controlled element.
    </>,
    <>
      Use <code>aria-describedby</code> prop to link the icon button to additional descriptive text. Pass the{" "}
      <code>id</code> of the descriptive element.
    </>,
    <>
      Use <code>aria-pressed</code> prop for toggle icon buttons to indicate their current pressed state (
      <code>true</code>, <code>false</code>, or <code>mixed</code>).
    </>,
    <>
      Use <code>aria-hidden</code> prop appropriately to hide the icon button from screen readers when necessary, but
      use sparingly as it removes the button from the accessibility tree entirely.
    </>,
    <>
      Use <code>tabIndex</code> prop to control the icon button's position in the keyboard navigation order. The default
      value allows normal tab navigation.
    </>,
    <>
      Provide <code>disabledReason</code> prop when disabling an icon button to give users context about why the button
      is unavailable. This will be displayed in the tooltip.
    </>
  ]}
/>

## Variants

### Kinds

There can be more than one button in a screen, but to create the hierarchy of actions we need to use button kinds.

<Canvas of={IconButtonStories.Kinds} />

<TipMenuButton />
### Sizes

<Canvas of={IconButtonStories.Sizes} />

### Active

<Canvas of={IconButtonStories.Active} />

### Disabled

Set disable button for something that isn’t usable. Use a tooltip on hover in order to indicate the reason of the disabled action.

<Canvas of={IconButtonStories.Disabled} />

## Do’s and Don’ts

<ComponentRules
  rules={[
    {
      positive: {
        component: (
          <DialogContentContainer>
            <IconButton icon={Pin} ariaLabel="Pin to top" />
            <IconButton icon={Show} ariaLabel="Show all" />
            <IconButton icon={HighlightColorBucket} ariaLabel="Show color" />
            <IconButton icon={Broom} ariaLabel="Clear style" />
          </DialogContentContainer>
        ),
        description:
          "Use Icon button when action is lower priority than a regular action or there’s no space available to place a button."
      },
      negative: {
        component: <IconButton icon={Add} kind="primary" ariaLabel="Add item" />,
        description: (
          <>
            Don’t use Icon button as the main action on the page. Instead, use the{" "}
            <StorybookLink page="Components/Button">Button component</StorybookLink> with an icon.
          </>
        )
      }
    },
    {
      positive: {
        component: <IconButton icon={Bolt} ariaLabel="Quick Search" />,
        description: "Always provide ariaLabel or tooltipContent"
      },
      negative: {
        component: <IconButton icon={Bolt} />,
        description: "Don’t use icon button without adding a tooltip while hovering."
      }
    }
  ]}
/>

## Use cases and examples

### Icon button as toolbar button

<Canvas of={IconButtonStories.IconButtonAsToolbarButton} />

### Icon button as close button

<Canvas of={IconButtonStories.IconButtonAsCloseButton} />

## Related components

<RelatedComponents componentsNames={["Button", "MenuButton", "Icon"]} />
