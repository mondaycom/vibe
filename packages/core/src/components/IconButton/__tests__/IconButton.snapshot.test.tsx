import { describe, it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import IconButton from "../IconButton";
import { SIZES } from "../../../constants/sizes";

describe("IconButton renders correctly", () => {
  it("with empty props", () => {
    const tree = renderer.create(<IconButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with id", () => {
    const tree = renderer.create(<IconButton id="icon-button" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with", () => {
    const tree = renderer.create(<IconButton className="class-name" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with disabled", () => {
    const tree = renderer.create(<IconButton disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`with size - xs`, () => {
    const tree = renderer.create(<IconButton size="xs" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`with size - medium`, () => {
    const tree = renderer.create(<IconButton size="medium" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  Object.values(SIZES).forEach(size => {
    it(`with size - ${size}`, () => {
      const tree = renderer.create(<IconButton size={size} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it("with aria label", () => {
    const tree = renderer.create(<IconButton ariaLabel="My aria label" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with active", () => {
    const tree = renderer.create(<IconButton active />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
