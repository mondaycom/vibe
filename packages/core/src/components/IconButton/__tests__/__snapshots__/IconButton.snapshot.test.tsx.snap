// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders correctly with 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="class-name button sizeMedium kindTertiary colorPrimary noSidePadding"
  data-testid="icon-button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    {
      "alignItems": "center",
      "height": "40px",
      "justifyContent": "center",
      "padding": 0,
      "width": "40px",
    }
  }
  type="button"
>
  <svg
    aria-hidden={true}
    className="icon noFocusStyle"
    data-testid="icon"
    data-vibe="Icon"
    fill="currentColor"
    height="20"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      clipRule="evenodd"
      d="M10.75 6C10.75 5.58579 10.4142 5.25 10 5.25C9.58579 5.25 9.25 5.58579 9.25 6V9.25H6C5.58579 9.25 5.25 9.58579 5.25 10C5.25 10.4142 5.58579 10.75 6 10.75H9.25V14C9.25 14.4142 9.58579 14.75 10 14.75C10.4142 14.75 10.75 14.4142 10.75 14V10.75H14C14.4142 10.75 14.75 10.4142 14.75 10C14.75 9.58579 14.4142 9.25 14 9.25H10.75V6Z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`IconButton renders correctly with active 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindTertiary colorPrimary colorPrimaryActive undefined noSidePadding"
  data-testid="icon-button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    {
      "alignItems": "center",
      "height": "40px",
      "justifyContent": "center",
      "padding": 0,
      "width": "40px",
    }
  }
  type="button"
>
  <svg
    aria-hidden={true}
    className="icon noFocusStyle"
    data-testid="icon"
    data-vibe="Icon"
    fill="currentColor"
    height="20"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      clipRule="evenodd"
      d="M10.75 6C10.75 5.58579 10.4142 5.25 10 5.25C9.58579 5.25 9.25 5.58579 9.25 6V9.25H6C5.58579 9.25 5.25 9.58579 5.25 10C5.25 10.4142 5.58579 10.75 6 10.75H9.25V14C9.25 14.4142 9.58579 14.75 10 14.75C10.4142 14.75 10.75 14.4142 10.75 14V10.75H14C14.4142 10.75 14.75 10.4142 14.75 10C14.75 9.58579 14.4142 9.25 14 9.25H10.75V6Z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`IconButton renders correctly with aria label 1`] = `
<span
  className="referenceWrapper"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button
    aria-busy={false}
    aria-disabled={false}
    aria-label="My aria label"
    className="button sizeMedium kindTertiary colorPrimary noSidePadding"
    data-testid="icon-button"
    data-vibe="Button"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    style={
      {
        "alignItems": "center",
        "height": "40px",
        "justifyContent": "center",
        "padding": 0,
        "width": "40px",
      }
    }
    type="button"
  >
    <svg
      aria-hidden={true}
      className="icon noFocusStyle"
      data-testid="icon"
      data-vibe="Icon"
      fill="currentColor"
      height="20"
      viewBox="0 0 20 20"
      width="20"
    >
      <path
        clipRule="evenodd"
        d="M10.75 6C10.75 5.58579 10.4142 5.25 10 5.25C9.58579 5.25 9.25 5.58579 9.25 6V9.25H6C5.58579 9.25 5.25 9.58579 5.25 10C5.25 10.4142 5.58579 10.75 6 10.75H9.25V14C9.25 14.4142 9.58579 14.75 10 14.75C10.4142 14.75 10.75 14.4142 10.75 14V10.75H14C14.4142 10.75 14.75 10.4142 14.75 10C14.75 9.58579 14.4142 9.25 14 9.25H10.75V6Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </button>
</span>
`;

exports[`IconButton renders correctly with disabled 1`] = `
<button
  aria-busy={false}
  aria-disabled={true}
  className="button sizeMedium kindTertiary colorPrimary noSidePadding disabled"
  data-testid="icon-button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    {
      "alignItems": "center",
      "height": "40px",
      "justifyContent": "center",
      "padding": 0,
      "width": "40px",
    }
  }
  tabIndex={-1}
  type="button"
>
  <svg
    aria-hidden={true}
    className="icon noFocusStyle"
    data-testid="icon"
    data-vibe="Icon"
    fill="currentColor"
    height="20"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      clipRule="evenodd"
      d="M10.75 6C10.75 5.58579 10.4142 5.25 10 5.25C9.58579 5.25 9.25 5.58579 9.25 6V9.25H6C5.58579 9.25 5.25 9.58579 5.25 10C5.25 10.4142 5.58579 10.75 6 10.75H9.25V14C9.25 14.4142 9.58579 14.75 10 14.75C10.4142 14.75 10.75 14.4142 10.75 14V10.75H14C14.4142 10.75 14.75 10.4142 14.75 10C14.75 9.58579 14.4142 9.25 14 9.25H10.75V6Z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`IconButton renders correctly with empty props 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindTertiary colorPrimary noSidePadding"
  data-testid="icon-button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    {
      "alignItems": "center",
      "height": "40px",
      "justifyContent": "center",
      "padding": 0,
      "width": "40px",
    }
  }
  type="button"
>
  <svg
    aria-hidden={true}
    className="icon noFocusStyle"
    data-testid="icon"
    data-vibe="Icon"
    fill="currentColor"
    height="20"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      clipRule="evenodd"
      d="M10.75 6C10.75 5.58579 10.4142 5.25 10 5.25C9.58579 5.25 9.25 5.58579 9.25 6V9.25H6C5.58579 9.25 5.25 9.58579 5.25 10C5.25 10.4142 5.58579 10.75 6 10.75H9.25V14C9.25 14.4142 9.58579 14.75 10 14.75C10.4142 14.75 10.75 14.4142 10.75 14V10.75H14C14.4142 10.75 14.75 10.4142 14.75 10C14.75 9.58579 14.4142 9.25 14 9.25H10.75V6Z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`IconButton renders correctly with id 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindTertiary colorPrimary noSidePadding"
  data-testid="icon-button_icon-button"
  data-vibe="Button"
  id="icon-button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    {
      "alignItems": "center",
      "height": "40px",
      "justifyContent": "center",
      "padding": 0,
      "width": "40px",
    }
  }
  type="button"
>
  <svg
    aria-hidden={true}
    className="icon noFocusStyle"
    data-testid="icon"
    data-vibe="Icon"
    fill="currentColor"
    height="20"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      clipRule="evenodd"
      d="M10.75 6C10.75 5.58579 10.4142 5.25 10 5.25C9.58579 5.25 9.25 5.58579 9.25 6V9.25H6C5.58579 9.25 5.25 9.58579 5.25 10C5.25 10.4142 5.58579 10.75 6 10.75H9.25V14C9.25 14.4142 9.58579 14.75 10 14.75C10.4142 14.75 10.75 14.4142 10.75 14V10.75H14C14.4142 10.75 14.75 10.4142 14.75 10C14.75 9.58579 14.4142 9.25 14 9.25H10.75V6Z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`IconButton renders correctly with size - large 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindTertiary colorPrimary noSidePadding"
  data-testid="icon-button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    {
      "alignItems": "center",
      "height": "48px",
      "justifyContent": "center",
      "padding": 0,
      "width": "48px",
    }
  }
  type="button"
>
  <svg
    aria-hidden={true}
    className="icon noFocusStyle"
    data-testid="icon"
    data-vibe="Icon"
    fill="currentColor"
    height="20"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      clipRule="evenodd"
      d="M10.75 6C10.75 5.58579 10.4142 5.25 10 5.25C9.58579 5.25 9.25 5.58579 9.25 6V9.25H6C5.58579 9.25 5.25 9.58579 5.25 10C5.25 10.4142 5.58579 10.75 6 10.75H9.25V14C9.25 14.4142 9.58579 14.75 10 14.75C10.4142 14.75 10.75 14.4142 10.75 14V10.75H14C14.4142 10.75 14.75 10.4142 14.75 10C14.75 9.58579 14.4142 9.25 14 9.25H10.75V6Z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`IconButton renders correctly with size - medium 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindTertiary colorPrimary noSidePadding"
  data-testid="icon-button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    {
      "alignItems": "center",
      "height": "40px",
      "justifyContent": "center",
      "padding": 0,
      "width": "40px",
    }
  }
  type="button"
>
  <svg
    aria-hidden={true}
    className="icon noFocusStyle"
    data-testid="icon"
    data-vibe="Icon"
    fill="currentColor"
    height="20"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      clipRule="evenodd"
      d="M10.75 6C10.75 5.58579 10.4142 5.25 10 5.25C9.58579 5.25 9.25 5.58579 9.25 6V9.25H6C5.58579 9.25 5.25 9.58579 5.25 10C5.25 10.4142 5.58579 10.75 6 10.75H9.25V14C9.25 14.4142 9.58579 14.75 10 14.75C10.4142 14.75 10.75 14.4142 10.75 14V10.75H14C14.4142 10.75 14.75 10.4142 14.75 10C14.75 9.58579 14.4142 9.25 14 9.25H10.75V6Z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`IconButton renders correctly with size - small 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindTertiary colorPrimary noSidePadding"
  data-testid="icon-button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    {
      "alignItems": "center",
      "height": "32px",
      "justifyContent": "center",
      "padding": 0,
      "width": "32px",
    }
  }
  type="button"
>
  <svg
    aria-hidden={true}
    className="icon noFocusStyle"
    data-testid="icon"
    data-vibe="Icon"
    fill="currentColor"
    height="20"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      clipRule="evenodd"
      d="M10.75 6C10.75 5.58579 10.4142 5.25 10 5.25C9.58579 5.25 9.25 5.58579 9.25 6V9.25H6C5.58579 9.25 5.25 9.58579 5.25 10C5.25 10.4142 5.58579 10.75 6 10.75H9.25V14C9.25 14.4142 9.58579 14.75 10 14.75C10.4142 14.75 10.75 14.4142 10.75 14V10.75H14C14.4142 10.75 14.75 10.4142 14.75 10C14.75 9.58579 14.4142 9.25 14 9.25H10.75V6Z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`IconButton renders correctly with size - xs 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindTertiary colorPrimary noSidePadding"
  data-testid="icon-button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    {
      "alignItems": "center",
      "height": "24px",
      "justifyContent": "center",
      "padding": 0,
      "width": "24px",
    }
  }
  type="button"
>
  <svg
    aria-hidden={true}
    className="icon noFocusStyle"
    data-testid="icon"
    data-vibe="Icon"
    fill="currentColor"
    height="16"
    viewBox="0 0 20 20"
    width="16"
  >
    <path
      clipRule="evenodd"
      d="M10.75 6C10.75 5.58579 10.4142 5.25 10 5.25C9.58579 5.25 9.25 5.58579 9.25 6V9.25H6C5.58579 9.25 5.25 9.58579 5.25 10C5.25 10.4142 5.58579 10.75 6 10.75H9.25V14C9.25 14.4142 9.58579 14.75 10 14.75C10.4142 14.75 10.75 14.4142 10.75 14V10.75H14C14.4142 10.75 14.75 10.4142 14.75 10C14.75 9.58579 14.4142 9.25 14 9.25H10.75V6Z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`IconButton renders correctly with size - xxs 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindTertiary colorPrimary noSidePadding"
  data-testid="icon-button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    {
      "alignItems": "center",
      "height": "16px",
      "justifyContent": "center",
      "padding": 0,
      "width": "16px",
    }
  }
  type="button"
>
  <svg
    aria-hidden={true}
    className="icon noFocusStyle"
    data-testid="icon"
    data-vibe="Icon"
    fill="currentColor"
    height="16"
    viewBox="0 0 20 20"
    width="16"
  >
    <path
      clipRule="evenodd"
      d="M10.75 6C10.75 5.58579 10.4142 5.25 10 5.25C9.58579 5.25 9.25 5.58579 9.25 6V9.25H6C5.58579 9.25 5.25 9.58579 5.25 10C5.25 10.4142 5.58579 10.75 6 10.75H9.25V14C9.25 14.4142 9.58579 14.75 10 14.75C10.4142 14.75 10.75 14.4142 10.75 14V10.75H14C14.4142 10.75 14.75 10.4142 14.75 10C14.75 9.58579 14.4142 9.25 14 9.25H10.75V6Z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
</button>
`;
